corehq/apps/case_search/utils.py

```diff
@@ -131,8 +131,13 @@ def wrap_case(self, es_hit, include_score=False):
         case.case_json[COMMCARE_PROJECT] = case.domain
         return case
 
-    def get_all_related_live_cases(self, cases):
-        return self.registry_helper.get_multi_domain_case_hierarchy(self.couch_user, cases)
+    def get_all_related_live_cases(self, initial_cases):
+        all_cases = self.registry_helper.get_multi_domain_case_hierarchy(self.couch_user, initial_cases)
+        initial_case_ids = {case.case_id for case in initial_cases}
+        result = list({
+            case.case_id: case for case in all_cases if case.case_id not in initial_case_ids
+        }.values())
```

Comment:
Is this filtering necessary if you are filtering again later?

Reply:
Ya I think so, otherwise this function wouldn't be returning only related cases as the function name implies. 

I guess the deduplication filtering isn't needed, and probably shouldn't since the same function for `_QueryHelper ` doesn't deduplicate. Is that what you're referring to? But I do think filtering out the `initial_cases` is needed.

Reply:
yea, I just meant filtering them out here vs in the caller.

Reply:
I removed the deduplication filtering from here [12e54ab](https://github.com/dimagi/commcare-hq/pull/32811/commits/12e54abef851804d99b274863a485e815d846a4a). 

I kept filtering out the `initial_cases`, i think it makes sense to keep that in implementation detail. But the caller `get_and_tag_related_cases` still needs to filter to account for `expanded_case_results`